src/Components/LowerComponenets/ArticleLIst.jsx

```diff
@@ -0,0 +1,66 @@
+import { useState, useContext, useEffect } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { Loading } from "./Loading";
+import { getArticles } from "../../Api";
+import { Error } from "./Error";
+
+export const ArticleList = () => {
+    const navigate = useNavigate();
+    const { topic } = useParams();
+    const [articles, setArticles] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState(null); 
+    
+    
+    useEffect(() => {
+        setIsLoading(true);
+        getArticles()
+        .then((articlesArrFromApi) => {
+            let filteredArticles = articlesArrFromApi;
+            if (topic) {
+                filteredArticles = articlesArrFromApi.filter((article) => article.topic === topic);
+            }
+            setArticles(filteredArticles);
+            setIsLoading(false);
+            })
+            .catch((error) => {
+                setError(error);
+                setIsLoading(false);
```

Comment:
There's no need to do the filtering in the front end. You set up queries in your back end that lets you filter articles by topic. For example the endpoint could be `/api/articles?topic=football` You can achieve this using the optional params object in axiom. Have a look at the axios notes in NC notes to help you with this 😄